### PR TITLE
HCO Upgrade test validates HCO CR condition statuses

### PR DESCRIPTION
**Pull request by MoserMichael**
upon completion of upgrade testing with make upgrade-test : check if cluster is up and running

Signed-off-by: MichaelMoser <mmoser@dhcp-2-163.tlv.redhat.com>

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by rthallisey** [comment]
ci add to whitelist

**Comment by rthallisey** [comment]
ci test please

File: hack/upgrade-test.sh [hunk 1]
```diff
@@ -211,22 +211,27 @@ ${CMD} wait pod $CATALOG_OPERATOR_POD --for condition=Ready -n openshift-operato
...
 echo "--"
 
 ./hack/retry.sh 6 30 "${CMD} get deployments -n kubevirt-hyperconverged -o yaml | grep image | grep hyperconverged-cluster | grep ${REGISTRY_IMAGE_URL_PREFIX}"
 
 echo "----- Images after upgrade"
 ${CMD} get deployments -n kubevirt-hyperconverged -o yaml | grep image | grep -v imagePullPolicy
 ${CMD} get pod $HCO_CATALOGSOURCE_POD -n ${HCO_CATALOG_NAMESPACE} -o yaml | grep image | grep -v imagePullPolicy
+
+echo "-- Upgrade Step 7/7: wait that cluster will become operational"
+make check-state
```
Comment (rwsu) [inline hack/upgrade-test.sh]: Can we also perform a check on statuses before the upgrade is started.

**Comment by rthallisey** [comment]
ci test please

**Comment by rthallisey** [comment]
lg2m.  @rwsu do you have any more comments?

**Comment by rwsu** [comment]
My comments above needs to be addressed. And this should go in after OpenShift CI is active.

File: Makefile [hunk 2]
```diff
@@ -90,6 +90,9 @@ dump-state:
 
 
  
+check-state:
+	timeout 10m bash -c 'export CMD="./cluster-up/kubectl.sh";exec ./hack/check-state.sh'
```
[the remark this answers is not in the document]
Comment (MoserMichael) [reply Makefile]: done.

File: hack/upgrade-test.sh (hunk 1 above)
Comment (MoserMichael) [reply hack/upgrade-test.sh]: done

File: hack/upgrade-test.sh [hunk 3]
```diff
@@ -181,8 +181,12 @@ echo "----- Images before upgrade"
 ${CMD} get deployments -n kubevirt-hyperconverged -o yaml | grep image | grep -v imagePullPolicy
 ${CMD} get pod $HCO_CATALOGSOURCE_POD -n ${HCO_CATALOG_NAMESPACE} -o yaml | grep image | grep -v imagePullPolicy
 
+#echo "-- check that cluster is operational"
+#timeout 10m bash -c 'export CMD="${CMD}";exec ./hack/check-state.sh'
```
Comment (rwsu) [inline hack/upgrade-test.sh]: Shouldn't this be enabled?

Comment (MoserMichael) [reply hack/upgrade-test.sh]: now enabled.

File: hack/upgrade-test.sh [hunk 4]
```diff
@@ -181,8 +181,12 @@ echo "----- Images before upgrade"
 ${CMD} get deployments -n kubevirt-hyperconverged -o yaml | grep image | grep -v imagePullPolicy
 ${CMD} get pod $HCO_CATALOGSOURCE_POD -n ${HCO_CATALOG_NAMESPACE} -o yaml | grep image | grep -v imagePullPolicy
 
+echo "-- check that cluster is operational"
```
Comment (rwsu) [inline hack/upgrade-test.sh]: Can you move lines 184-185 to right after line 175?

**Comment by MoserMichael** [comment]
> The CI results are showing that the status condition checks is being skipped. @MoserMichael can you take alook?
> 
> Example: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/336/pull-ci-kubevirt-hyperconverged-cluster-operator-master-hco-e2e-upgrade-aws/91/artifacts/hco-e2e-upgrade-aws/container-logs/test.log
> 
> ```
> Checking operator kind: hyperconvergeds.hco.kubevirt.io name: hyperconverged-cluster 
> Operator hyperconvergeds.hco.kubevirt.io doesn't support required status conditions. Skipping check.
> ```

Done.

File: hack/upgrade-test.sh (hunk 4 above)
Comment (MoserMichael) [reply hack/upgrade-test.sh]: done.

Comment (MoserMichael) [reply hack/upgrade-test.sh]: done

**Pull request merged**